### PR TITLE
Fix issues introduced by the benchmarking mode changes

### PR DESCRIPTION
**Pull request by Adamantios**
Fixes the issues introduced in #347 and replaces #362.

### CODE REVIEW

File: packages/valory/skills/decision_maker_abci/behaviours/sampling.py [hunk 1]
```diff
@@ -151,8 +151,8 @@ def async_act(self) -> Generator:
         """Do the action."""
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
             idx = self._sample()
-            benchmarking_finished = False
-            day_increased = False
+            benchmarking_finished = None
+            day_increased = None
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/sampling.py:155]: This was causing an issue in the **normal flow**.

File: packages/valory/skills/decision_maker_abci/rounds.py [hunk 2]
```diff
@@ -218,6 +219,7 @@ class DecisionMakerAbciApp(AbciApp[Event]):
             Event.DONE: SubscriptionRound,
             Event.NONE: FinishedWithoutDecisionRound,
             Event.NO_MAJORITY: SamplingRound,
+            Event.ROUND_TIMEOUT: SamplingRound,
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/rounds.py:222]: This was causing an issue in the **normal flow**.

File: packages/valory/skills/decision_maker_abci/states/sampling.py [hunk 3]
```diff
@@ -48,6 +48,7 @@ class SamplingRound(UpdateBetsRound):
         get_name(SynchronizedData.benchmarking_finished),
         get_name(SynchronizedData.simulated_day),
     )
+    synchronized_data_class = SynchronizedData
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/states/sampling.py:51]: This was causing an issue in the **normal flow**.

**Review by keshav1998 — APPROVED** [review]

**Review by cyberosa — APPROVED** [review]

File: packages/valory/skills/decision_maker_abci/behaviours/sampling.py (hunk 1 above)
Comment (cyberosa) [reply packages/valory/skills/decision_maker_abci/behaviours/sampling.py:155]: Still trying to understand why None is a valid value and False not.

File: packages/valory/skills/decision_maker_abci/states/sampling.py (hunk 3 above)
Comment (Adamantios) [reply packages/valory/skills/decision_maker_abci/states/sampling.py:51]: Can you elaborate on that?

File: packages/valory/skills/decision_maker_abci/behaviours/sampling.py (hunk 1 above)
Comment (Adamantios) [reply packages/valory/skills/decision_maker_abci/behaviours/sampling.py:155]: Because otherwise the NONE event cannot ever be emitted from the round.

**Pull request merged**